packages/ketcher-react/src/script/ui/data/convert/structConverter.ts

```diff
@@ -190,46 +160,54 @@ export function fromString(structStr, opts, server, serverOpts) {
   })
 }
 
-export function couldBeSaved(struct, format) {
-  let warnings = []
+export function couldBeSaved(
+  struct: any,
+  format: SupportedFormat
+): string | null {
+  let warnings: Array<string> = []
+  const formatName: string = getPropertiesByFormat(format).name
   if (
-    format === 'inchi' ||
-    format === 'inchi-aux' ||
-    format === 'smiles' ||
-    format === 'smiles-ext'
+    format === SupportedFormat.InChI ||
+    format === SupportedFormat.InChIAuxInfo ||
+    format === SupportedFormat.Smiles ||
+    format === SupportedFormat.SmilesExt
   ) {
     if (struct.rgroups.size !== 0)
       warnings.push(
-        `In ${map[format].name} the structure will be saved without R-group fragments`
+        `In ${formatName} the structure will be saved without R-group fragments`
       )
+
     struct = struct.clone() // need this: .getScaffold()
+    // @ts-ignore
     const isRg = struct.atoms.find((ind, atom) => atom.label === 'R#')
     if (isRg !== null)
       warnings.push(
-        `In ${map[format].name} the structure will be saved without R-group members`
+        `In ${formatName} the structure will be saved without R-group members`
       )
 
     const isSg = struct.sgroups.find(
+      // @ts-ignore
       (ind, sg) =>
         sg.type !== 'MUL' && !/^INDIGO_.+_DESC$/i.test(sg.data.fieldName)
     )
     if (isSg !== null)
       warnings.push(
-        `In ${map[format].name} the structure will be saved without S-groups`
+        `In ${formatName} the structure will be saved without S-groups`
       )
   }
 
   if (
-    format === 'smiles' ||
-    format === 'smiles-ext' ||
-    format === 'smarts' ||
-    format === 'inchi' ||
-    format === 'inchi-aux' ||
-    format === 'cml'
+    format === SupportedFormat.Smiles ||
+    format === SupportedFormat.SmilesExt ||
+    format === SupportedFormat.Smarts ||
+    format === SupportedFormat.InChI ||
+    format === SupportedFormat.InChIAuxInfo ||
+    format === SupportedFormat.CML
   ) {
```

Comment:
I'd prefer to use `switch` statement instead of two `if`, when we just compare variable value with several variants

Reply:
+ row #173

Reply:
fixed both